app/services/recommendation_service.rb

```diff
@@ -8,57 +8,70 @@
-    # sort true with Euclidean Distance, false with cosine similarity
-    max = similar.loc['Max'].sort_values(0,ascending=true)[1..5] #here, Max is replacing our USER_ID for the user requesting the rec
-    # max = similar.loc['Max'].sort_values(0,ascending=false)[1..5] #here, Max is replacing our USER_ID for the user requesting the rec
-    df_max = Pandas.DataFrame.new(data=max, index=df.index)
-
-    # prepping the cocktail data to merge with the user similarity data
-    pivoted = Pandas.melt(df.reset_index(),id_vars='name',value_vars=df.keys)
-
-    # scraping out 0 (nil) ratings and merging similarity data with cocktail data
-    scraped_pivot = pivoted[pivoted.value != 0]
-    total = df_max.reset_index().merge(scraped_pivot).dropna()
-
-    # creating a comparison metric (weighted rating) based on euclidean distance (ed_adjusted * rating)
-    total['weightedRating']=(1 / total['Max']+1)*total['value']
-    # total['weightedRating']=total['Max']*total['value'] # Use this one for cosine similarity
-
-    # Adding up the similarities by drink and aggregating the data
-    similarity_weighted = total.groupby('variable').sum()[['Max','weightedRating']]
-    
-    # Removing drinks user has already rated
-    max_pivot = pivoted[pivoted.name == 'Max'] # won't be 'Max', will be user_id (current_user.id?)
-    reset_sw = similarity_weighted.reset_index()
-    unknown_to_user = reset_sw.merge(max_pivot).set_index('variable')
-    unknown_to_user = unknown_to_user[unknown_to_user.value == 0]
-
-    # Finishing created the weighted avg adjusted euclid distance
-    empty = Pandas.DataFrame.new()
-    empty['weightedAvgRecScore'] = unknown_to_user['Max']/unknown_to_user['weightedRating']
-
-    # Make the final (best) recommendation
-    recommendation = empty[empty.weightedAvgRecScore == empty.weightedAvgRecScore.max()]
-    # empty.loc[empty['weightedAvgRecScore'].idxmax()]
-
-    # need to format for incoming data with user_ids, etc. 
-    require 'pry'; binding.pry
+    user_distance_matrix = Pandas.DataFrame.new(data=euclidean, index=df.index,columns=df.index)
+    # binding.pry
+  end
+
+  # def self.create_df
+  #   csv_data = Pandas.read_csv("./db/data/cocktail_ratings.csv") #replace with our ratings data
+  #   binding.pry
+  #   df = csv_data.set_index('name').fillna(0) #will be user_id instead of name
+  # end
+
+  def self.recommendation(user_id)
+    # df = create_df
+    df = Dataframeable.create_df
+
+    user_distance_matrix = euclidean_distance(df)
+
+    distances_from_user = user_distances(user_distance_matrix, df, user_id)
+
+    cocktail_ratings = Pandas.melt(df.reset_index(),id_vars='user_id',value_vars=df.keys)
+    scraped_ratings = cocktail_ratings[cocktail_ratings.value != 0]
+
+    baseline_ratings = distances_from_user.reset_index().merge(scraped_ratings).dropna()
+
+    weighted_ratings(baseline_ratings, user_id)
+
+    combined_ratings = baseline_ratings.groupby('cocktail_id').sum()[[user_id,'weightedRating']]
+
```

Comment:
Maybe if we put all 4 of these together we could pull them out into another method? Because we need cocktail ratings and combined ratings externally, but I don't think we need the others later (?)